pypiserver/__main__.py

```diff
@@ -84,25 +208,52 @@ def main(argv=None):
     from pypiserver import bottle
 
     bottle.debug(config.verbosity > 1)
-    bottle._stderr = ft.partial(
+    bottle._stderr = ft.partial(  # pylint: disable=protected-access
         _logwrite, logging.getLogger(bottle.__name__), logging.INFO
     )
 
     # Here `app` is a Bottle instance, which we pass to bottle.run() to run
     # the server
     app = pypiserver.app_from_config(config)
+
+    if config.server_method == "gunicorn":
+        # When bottle runs gunicorn, gunicorn tries to pull its arguments from
+        # sys.argv. Because pypiserver's arguments don't match gunicorn's,
+        # this leads to errors.
+        sys.argv = ["gunicorn"]
```

Comment:
How can we pass additional configuration to gunicorn? only through a `gunicorn.conf.py`?

And just out of curiosity: how is the `gunicorn.conf.py` passed as a config file in the docker container? Or does `gunicorn` pick it up automatically from the user's $HOME?

Reply:
Yeah, this is another thing I struggled with. There are a couple of ways as it stands that folks could configure gunicorn. One is setting the `GUNICORN_CMD_ARGS` env var, another is by mounting a new gunicorn config file, and another would be by setting certain env vars directly. But, I couldn't figure out any sensible way to have them passed via the commandline.

The `gunicorn.conf.py` is picked up automatically when it's in the current working directory when you're running gunicorn. We might consider adding something like a `--gunicorn-conf` option to allow people to specify some config file rather than using gunicorn's default search mechanisms, but since they can already do this with `GUNICORN_CMD_ARGS="-c my_config_file.py"` it seemed like it wasn't strictly necessary

Reply:
Ah right, it's automatically picked up from the working dir, I had a look at the gunicorn docs, but couldn't find it there. Thanks :)

I'd say adding a gunicorn specific command line argument is not really the way to go. Maybe in the future there could be a way to pass arbitrary options to arbitrary servers, but I guess that's for another time :)

Reply:
I threw a few extra comments in around this in the most recent commits. It'll also be something we want to document as we work on getting the docs up to snuff for v2